### ISSUE TITLE
Page refresh in zoom in-out event  Mapbox

### ISSUE THREAD

**Issue by achalnexus** [mapbox/mapbox-gl-js#2995]
[Mapbox.txt](https://github.com/mapbox/mapbox-gl-js/files/415743/Mapbox.txt)
I have mapbox implement in my asp.net project i have create sample demo but i face zoom-out issue.
If i am click on zoom-out button then after automatic refresh page. i don't want to page refresh on zoom-out event using from tag. from tag must be use in asp.net.

<<<<<<<<<<============= My ASP.Net Page Code ==========>>>>>>>>>>>>>>>

<%@ Page Language="C#" AutoEventWireup="true" CodeFile="Default4.aspx.cs" Inherits="Default4" %>

<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head runat="server">
    <title></title>
    <meta name='viewport' content='initial-scale=1,maximum-scale=1,user-scalable=no' />
    <script src="http://ajax.aspnetcdn.com/ajax/jQuery/jquery-1.12.4.min.js"></script>
    <script src='https://api.tiles.mapbox.com/mapbox-gl-js/v0.21.0/mapbox-gl.js'></script>
    <link href='https://api.tiles.mapbox.com/mapbox-gl-js/v0.21.0/mapbox-gl.css' rel='stylesheet' />
    <style>
        body {
            margin: 0;
            padding: 0;
        }
        #map {
            position: absolute;
            top: 0;
            bottom: 0;
            width: 100%;
        }
        .mapboxgl-ctrl-compass {
            display: none !important;
        }
        #fly {
            display: block;
            position: relative;
            margin: 0px auto;
            width: 10%;
            height: 40px;
            padding: 10px;
            border: none;
            border-radius: 3px;
            font-size: 12px;
            text-align: center;
            color: #fff;
            background: #000;
        }
    </style>
</head>
<body>
    <form id="form1" runat="server">
        <div id='map'></div>
        <button id='fly'>Search this area</button>
    </form>
    <script src="GLMapbox.js"></script>
    <script>
        $(document).ready(function () {
            Init();
        });
    </script>
</body>
</html>

 <<<<<<<<<<============= My Mapbox Script Code ==========>>>>>>>>>>>>>>>

mapboxgl.accessToken = 'pk.eyJ1IjoiYWNoYWxwcmFqYXBhdGkiLCJhIjoiY2lyNGkwZGsxMDFpenUybHd5bjRtMjVjeiJ9.2teTa5MmVqOW-MDpryv56w';
function Init() {
    if (!mapboxgl.supported()) {
        console.log('Mapbox GL not supported! Using standard map');
    } else {
        var center = [-94.6858998, 46.729553];
        var map = new mapboxgl.Map({
            container: 'map', // container id or map id
            style: 'mapbox://styles/achalprajapati/cir4j4zcq000gcgnbu8t4k8wf', //stylesheet location
            center: center, // starting position
            zoom: 9, // starting zoom
            //position: 'top-left',
            minZoom: 7,
            maxZoom: 15,
            keyboard: true,
            doubleClickZoom: false,
        });
        map.scrollZoom.disable(); // Drage n drop 
        map.addControl(new mapboxgl.Navigation({ position: 'top-left' })); // Zoom in-out navigation control//
        map.on('style.load', function () {
            map.addSource("markers", {
                "type": "geojson",
                "data": {
                    "type": "FeatureCollection",
                    "features": [{
                        "type": "Feature",
                        "geometry": {
                            "type": "Point",
                            "coordinates": [-94.6858998, 46.729553]
                        },
                        "properties": {
                            "title": "Afton State Park",
                            "description": "<strong>Make it Mount Pleasant</strong><p><a href=\"javascript:void(0);\"  onclick=\'alert();\' title=\"Opens in a new window\">Make it Mount Pleasant</a></p>",
                            "marker-symbol": "ap"
                        }
                    }, {
                        "type": "Feature",
                        "geometry": {
                            "type": "Point",
                            "coordinates": [-95.8903015, 44.3860723]
                        },
                        "properties": {
                            "title": "Camden State Park",
                            "marker-symbol": "mapload"
                        }
                    }]
                }
            });

            map.addLayer({
                "id": "markers",
                "type": "symbol",
                "source": "markers",
                "layout": {
                    "icon-image": "{marker-symbol}-15",
                    "text-field": "{title}",
                    "text-font": ["Open Sans Semibold", "Arial Unicode MS Bold"],
                    "text-offset": [0, 0.3],
                    "text-anchor": "top"
                }
            });
        });
        map.on('zoomstart', function (e) {
            alert(map.getZoom());
        });
        map.on('zoomend', function (e) {
            alert(map.getZoom());
        });      
    }
}

**Comment by lucaswoj** [comment]
Thanks for getting in touch @achalnexus. This issue is a duplicate of #2934, which has been fixed in the latest release.

**Comment by achalnexus** [comment]
Give me demo or link for example and open your link but not open any source code.
